design/experimental-new-features-in-modular.md

```diff
@@ -0,0 +1,204 @@
+
+1. **Version picking strategy**  
+  One latest GA + One latest preview version.  
+  If the latest preview version < latest GA version, then we will use latest GA version as the experiment version.  
+  _Questions_:
+    1. do we allow customers to set their own preview version?
+
+1. **Feature picking strategy**  
+  Let's say we have Feature 1, 2, 3, 4 to experiment
+  _Questions_:
+    1. Is it possible that we want features 1, 2 in one RP but features 3, 4 in another RP?
+    1. If service team want a pure preview with their service version without any client side preview features, is this allowed?  
+
+1. **Feature staging status change**  
+  _Questions_:
+   1. Is it possible that a feature could be GA or withdrew based on customers feedback?
+   1. how are we going to decide, statistically?
+   1. If a feature goes GA, we shall refresh all the packages that has the experimented feature, or just refresh all the packages based on release requirement?
+   1. If a feature goes withdraw, we shall refresh all the packages that has the experiment feature?
+
+1. **Feature collisions**
+   1. Is it possible that one feature may have part of the code collisions with the other feature?
+   1. Is it possible that experimental features may have part of the code collisions with the stable code?
+   1. What if a feature requires a more newer version of dependencies libraries ? like a newer version of typescript or azure core related libraries?  
+     Should we use something like the below if such case happens ? or should we just use the latest?
+
+   ```json
+   "dependencies": {
+      "@azure-rest/map-search-v1": "npm:@azure-rest/map-search@^1.0.0",
+      "@azure-rest/map-search-v2": "npm:@azure-rest/map-search@^2.0.0"
+    } 
+   ```
+
+1. **Sub client picking strategy**  
+In the multi-client and multi-endpoint cases,  
+_Questions_:
+   1. is it possible that we could only select one sub client to ship with experimental features?
+
+1. **Manual customization**  
+    1. Is it allowed to have manual customization in the experimental part?
+    1. Is this possible that we may have a customization scenario that involves both the stable code and experimental code?
+
+
+1. **SDK version strategy**  
+  Should we allow pure experimental feature releases?  
+    1. a preview api version  --- 1.0.0-beta.1
+    1. ~~a new experimental feature --- 1.0.0-beta.2~~  
+    1. a new preview api version --- 1.0.0-beta.3  
+    1. ~~a new experimental features --- 1.0.0-beta.4~~  
+    1. a new stable api version --- 1.0.0 will need to replace  
+    1. ~~a new experimental features --- 1.1.0-beta.1 ?~~
+    1. a new preview api version ---  1.1.0-beta.2 ?
+
+## Design Considerations
+
+1. Experimental features may have changes over the original code structure.
+1. Experimental features may need to change some common dependencies.
+1. Api version picking strategy: latest GA version go to the stable code and latest preview version go to the vnext code.
```

Comment:
This sounds to me like it's suggesting something implicit, like

"IF the user selects a preview version THEN they will get all experimental features automatically", where I think we may really want the reverse: "IF a user selects experimental features, THEN they must use the preview version that supports them..."